components/support/locale/src/main/java/mozilla/components/support/locale/LocaleMiddleware.kt

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.locale
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.launch
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.action.LocaleAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import mozilla.components.lib.state.Store
+import mozilla.components.support.base.log.logger.Logger
+import java.util.Locale
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * [Middleware] implementation for updating [BrowserState.locale] state changes during restore.
+ */
+class LocaleMiddleware(
+    private val applicationContext: Context,
+    coroutineContext: CoroutineContext = Dispatchers.IO,
+    private val localeManager: LocaleManager,
+    private val localeUseCases: LocaleUseCases
```

Comment:
I'm not sure if this is the right way to access these

Reply:
You can omitted the `LocaleUseCases` as you can use the `store` to `dispatch`. The use case are shortcuts :)

Reply:
you could also pass a reference of the `LocaleManager` directly to the middle ware